### PR TITLE
Export elements data

### PR DESCRIPTION
**Pull request by deadlyfingers**
Unwinds query and data results for each element, including elements within dialogs.
NB: The datasource id & element id need to match for this to happen.

### CODE REVIEW

**Comment by deadlyfingers** [comment]
@morsh as part of this PR do you want me to update the preconfigured template files to make sure they work with the export feature? Only some of the id names need to be changed.

File: client/src/data-sources/plugins/ApplicationInsights/Query.ts [hunk 1]
```diff
@@ -164,6 +164,22 @@ export default class ApplicationInsightsQuery extends DataSourcePlugin<IQueryPar
     }
   }
 
+  getElementQuery(dataSource: IDataSource, dependencies: IDict<any>, aQuery: string, queryFilters: any): string {
```
Comment (morsh) [inline client/src/data-sources/plugins/ApplicationInsights/Query.ts]: What is `aQuery` - name should be descriptive.

File: client/src/data-sources/plugins/ApplicationInsights/Query.ts [hunk 2]
```diff
@@ -164,6 +164,22 @@ export default class ApplicationInsightsQuery extends DataSourcePlugin<IQueryPar
...
+    let timespan = '30d';
+    const table = dataSource['config'].params.table || null;
+    if (dependencies['timespan'] && dependencies['timespan']['queryTimespan']) {
```
Comment (morsh) [inline client/src/data-sources/plugins/ApplicationInsights/Query.ts]: Great, but add another check on `dependencies` itself

File: client/src/data-sources/plugins/ApplicationInsights/Query.ts [hunk 3]
```diff
@@ -164,6 +164,22 @@ export default class ApplicationInsightsQuery extends DataSourcePlugin<IQueryPar
     }
   }
 
+  getElementQuery(dataSource: IDataSource, dependencies: IDict<any>, aQuery: string, queryFilters: any): string {
+    let timespan = '30d';
+    const table = dataSource['config'].params.table || null;
```
Comment (morsh) [inline client/src/data-sources/plugins/ApplicationInsights/Query.ts]: const table = dataSource && dataSource['config'] && dataSource['config'].params && dataSource['config'].params.table || null;

**Pull request merged**